pkg/definition/gen_sdk/_scaffold/go/pkg/apis/common/application.go

```diff
@@ -0,0 +1,299 @@
+	var result []Policy
+	for _, p := range a.policies {
+		if p.DefType() == typ {
+			result = append(result, p)
+		}
+	}
+	return result
+}
+
+func (a *ApplicationBuilder) WithWorkflowSteps(step ...WorkflowStep) Application {
+	a.steps = append(a.steps, step...)
+	return a
+}
+
+// WithComponents append components to application
+func (a *ApplicationBuilder) WithComponents(component ...Component) Application {
+	a.components = append(a.components, component...)
+	return a
+}
+
+// WithPolicies append policies to application
+func (a *ApplicationBuilder) WithPolicies(policy ...Policy) Application {
+	a.policies = append(a.policies, policy...)
+	return a
+}
+
+// WithWorkflowMode set the workflow mode of application
+func (a *ApplicationBuilder) WithWorkflowMode(steps, subSteps common.WorkflowMode) Application {
+	a.workflowMode.Steps = steps
+	a.workflowMode.SubSteps = subSteps
+	return a
+}
+
+func (a *ApplicationBuilder) Name(name string) Application {
+	a.name = name
+	return a
+}
+
+func (a *ApplicationBuilder) Namespace(namespace string) Application {
+	a.namespace = namespace
+	return a
+}
+
+func (a *ApplicationBuilder) Labels(labels map[string]string) Application {
+	a.labels = labels
+	return a
+}
+
+func (a *ApplicationBuilder) Annotations(annotations map[string]string) Application {
+	a.annotations = annotations
+	return a
+}
+
+func (a *ApplicationBuilder) GetName() string {
+	return a.name
+}
+
+func (a *ApplicationBuilder) GetNamespace() string {
+	return a.namespace
+}
+
+func (a *ApplicationBuilder) GetLabels() map[string]string {
+	return a.labels
+}
+
+func (a *ApplicationBuilder) GetAnnotations() map[string]string {
+	return a.annotations
+}
+
+// New creates a new application with the given components.
+func New() Application {
+	app := &ApplicationBuilder{
+		components: make([]Component, 0),
+		steps:      make([]WorkflowStep, 0),
+		policies:   make([]Policy, 0),
+	}
+	return app
+}
+
+func (a *ApplicationBuilder) Build() v1beta1.Application {
+	components := make([]common.ApplicationComponent, 0, len(a.components))
+	for _, component := range a.components {
+		components = append(components, component.Build())
+	}
+	steps := make([]v1beta1.WorkflowStep, 0, len(a.steps))
+	for _, step := range a.steps {
+		steps = append(steps, step.Build())
+	}
+	policies := make([]v1beta1.AppPolicy, 0)
+	for _, policy := range a.policies {
+		policies = append(policies, policy.Build())
+	}
+
+	res := v1beta1.Application{
+		TypeMeta: v1.TypeMeta{
+			Kind:       v1beta1.ApplicationKind,
+			APIVersion: v1beta1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:            a.name,
+			Namespace:       a.namespace,
+			ResourceVersion: a.resourceVersion,
+		},
+		Spec: v1beta1.ApplicationSpec{
+			Components: components,
+			Workflow: &v1beta1.Workflow{
+				Steps: steps,
+			},
+			Policies: policies,
+		},
+	}
+	return res
+}
+
+func FromK8sObject(app *v1beta1.Application) (Application, error) {
```

Comment:
what's the difference between `v1beta1.Application` and `Application` ?

Reply:
`Application` is the internal type. It holds typed component/trait/... with typed properties. `v1beta1.Application` can not hold typed component.

Reply:
I got, but why not a better name. TypedApplication is obviously better.